jgiven-core/src/test/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportBlockConverterTest.java

```diff
@@ -0,0 +1,647 @@
+            "+++Best scenario ever!!!+++");
+    }
+
+    @Test
+    public void convert_scenario_header_with_tags_and_description() {
+        // arrange
+        List<String> tagNames = new ArrayList<>();
+        tagNames.add("Best Tag");
+
+        // act
+        String block =
+            converter.convertScenarioHeaderBlock("my first scenario", ExecutionStatus.SUCCESS, 3000000000L, tagNames,
+                "Best scenario ever!!!");
+
+        // assert
+        assertThatBlockContainsLines(block,
+            "// tag::scenario-successful[]",
+            "",
+            "==== My first scenario",
+            "",
+            "icon:check-square[role=green] (3s 0ms)",
+            "",
+            "+++Best scenario ever!!!+++",
+            "",
+            "Tags: _Best Tag_");
+    }
+
+    @Test
+    @DataProvider({"SUCCESS, check-square[role=green]",
+        "FAILED, exclamation-circle[role=red]",
+        "SCENARIO_PENDING, ban[role=silver]",
+        "SOME_STEPS_PENDING, ban[role=silver]"})
+    public void convert_case_header_without_parameters_and_description(final ExecutionStatus executionStatus,
+                                                                       final String icon) {
+        // arrange
+
+        // act
+        String block = converter.convertCaseHeaderBlock(1, executionStatus, null);
+
+        // assert
+        assertThatBlockContainsLines(block,
+            "===== Case 1",
+            "",
+            "icon:" + icon);
+    }
+
+    @Test
+    public void convert_case_header_with_description_and_without_parameters() {
+        // arrange
+
+        // act
+        String block =
+            converter.convertCaseHeaderBlock(1, ExecutionStatus.SUCCESS, "First case");
+
+        // assert
+        assertThatBlockContainsLines(block,
+            "===== Case 1 First case",
+            "",
+            "icon:check-square[role=green]");
+    }
+
+    @Test
+    public void convert_case_header_with_one_parameter() {
+        // arrange
+
+        // act
+        String block =
+            converter.convertCaseHeaderBlock(2, ExecutionStatus.SUCCESS, null);
+
+        // assert
+        assertThatBlockContainsLines(block,
+            "===== Case 2",
+            "",
+            "icon:check-square[role=green]");
+    }
+
+    @Test
+    public void convert_case_header_with_two_parameters() {
+        // arrange
+
+        // act
+        String block =
+            converter.convertCaseHeaderBlock(2, ExecutionStatus.SUCCESS, null);
+
+        // assert
+        assertThatBlockContainsLines(block,
+            "===== Case 2",
+            "",
+            "icon:check-square[role=green]");
+    }
+
+    @Test
+    public void convert_step_without_description() {
+        // arrange
+        List<Word> words = ImmutableList.of(Word.introWord("given"), new Word("a coffee machine"));
+
+        // act
+        String block = converter.convertStepBlock(0, words, StepStatus.PASSED, 3899, null, false);
```

Comment:
I would put the number `3899` in a global constant name `DURATION` (or DURATION_IN_NANOS). The number doesn't actually matter, but a reader would have an easier time determining its significance